com.microsoft.java.lsif.core/src/com/microsoft/java/lsif/core/internal/indexer/Repository.java

```diff
@@ -90,6 +109,43 @@ public synchronized SymbolData enlistSymbolData(String id, Document docVertex, P
 		return symbolData;
 	}
 
+	public synchronized PackageInformation enlistPackageInformation(LsifService lsif, String id, String name,
+			PackageManager manager, String version, String type, String url) {
+		PackageInformation packageInformation = findPackageInformationById(id);
+		if (packageInformation == null) {
+			if (StringUtils.isEmpty(name)) {
+				return null;
+			}
+			packageInformation = lsif.getVertexBuilder().packageInformation(name, manager, version, type, url);
+			addPackageInformation(id, packageInformation);
+			LsifEmitter.getInstance().emit(packageInformation);
+		}
+		return packageInformation;
+	}
+
+	public synchronized MavenProject enlistMavenProject(LsifService lsif, File pomFile) {
+		MavenProject mavenProject = findMavenProjectByPath(pomFile.getAbsolutePath());

```

Comment:
This seems not correct. Imagine all the thread will waiting here for resolving the maven project.

IMO, this class should only do simply read-write task to avoid becoming perf bottle-neck